builtin/credential/cert/path_certs.go

```diff
@@ -132,6 +133,22 @@ func (b *backend) pathCertWrite(
 		return logical.ErrorResponse("failed to parse certificate"), nil
 	}
 
+	// If the certificate is not a CA cert, then ensure that x509.ExtKeyUsageClientAuth is set
+	if !parsed[0].IsCA {
+		if parsed[0].ExtKeyUsage == nil {
+			return logical.ErrorResponse("non-CA certificates should have TLS web client authentication set as an extended key usage"), nil
```

Comment:
Also, if ExtKeyUsage isn't set, it actually defaults to any allowed usage. It's a restriction list, not an allow list. So if it's *not* nil is when you need to check.